### PR TITLE
feat: filtering wrappers

### PR DESCRIPTION
**Pull request by AgAriel**
Added filtering functions for `[]string` and `[][]string`

### CODE REVIEW

File: slice.go [hunk 1]
```diff
@@ -29,3 +31,22 @@ func NoLast(ls ...string) []string {
...
+type predicate func(string) bool
+type predicateLs func([]string) bool
+
+// Filter filters `ls` according to `p`
+func Filter(ls []string, p predicate) []string {
+	return funk.FilterString(ls, p)
+}
+
+// SliceFilter filters `ls` according to `p`
+func SliceFilter(ls [][]string, p predicateLs) [][]string {
```
Comment (ferhatelmas) [inline slice.go]: `FilterSlice` sounds better to me for a func name

File: slice.go [hunk 2]
```diff
@@ -29,3 +31,22 @@ func NoLast(ls ...string) []string {
 	}
 	return ls[:len(ls)-1]
 }
+
+type predicate func(string) bool
```
Comment (ferhatelmas) [inline slice.go]: export since they are expected in the signatures

File: slice.go [hunk 3]
```diff
@@ -29,3 +31,22 @@ func NoLast(ls ...string) []string {
...
+type predicateLs func([]string) bool
```
Comment (ferhatelmas) [inline slice.go]: `ls` is more for python, let's use slice

**Comment by ferhatelmas** [comment]
put them into a separate file, i.e. _filter.go_

File: slice.go (hunk 1 above)
Comment (AgAriel) [reply slice.go]: Yeah I agree with the name, i just put these according to `contains` prefix convention, without anything will be for a []string, `String` will mean for a string and `Slice` for a [][]string, so is it ok to rename it?

**Comment by AgAriel** [comment]
> put them into a separate file, i.e. _filter.go_

I was putting on slice.go because on that file there are the subslices functions like Tail and NoLast, like this is a kind of that I was putting there what do yo think?

**Comment by ferhatelmas** [comment]
> I was putting on slice.go because on that file there are the subslices functions like Tail and NoLast, like this is a kind of that I was putting there what do yo think?

thanks for explanation but IMHO, filter is still better.

**Pull request merged**